### DISCUSSION TITLE
Page Background Color

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#2276] [Looking for help] [answered]
Can PyMuPDF extract background color of a page or text?

**Comment by JorjMcKie** [comment]
No, because you cannot define this in a unique way, can you?

But you can create a pixmap of the bbox of some text and check whether it is unicolor (only one color), whether there is a predominant color: `pix.color_topusage()`. If this returns that for example 95% of the area is colored the same way, then the text will probably not be readable.

**Reply by meghanaviyyapu** [reply]
Ok thank you

**Comment by meghanaviyyapu** [comment, marked as answer]
Can you let me know how a pixmap of bbox of text can be created using PyMuPDF?

**Reply by JorjMcKie** [reply]
I assume you want to check whether some given text is actually visible, right?
First thing you need to know is the `bbox` (boundary box) of that text.

Then do `pix = page.get_pixmap(clip=bbox)`. As mentioned above you now can check `pix.is_unicolor`. If true, the text is definitely invisible.

To lower this very strict condition, you can alternatively compute `top_color = pix.color_topusage()`. This a tuple `(ratio, pixel)`, where ratio is a float 0 < ratio <= 1 indicating a relative frequency of color `pixel`.
`pixel` is a bytes object representing a pixel value. If ratio is close to 1, then "almost all" pixels have a color `pixel`, which would mean that the text is "almost" invisible.
If ratio == 1, then this is equivalent to being unicolor.

> If you need to check many text bboxes **_on the same page_**, it makes sense performance-wise to create the pixmap of the full page once only, and then check `pix.color_topusage(clip=bbox)` for each of the bboxes. This approach avoids computing multiple pixmaps (which can be time consuming).

**Reply by meghanaviyyapu** [reply]
I want to extract the background color of  text with bbox value. Can you let me know if the color of background of selected bbox can be extracted?

**Reply by JorjMcKie** [reply]
Same answer: there is no such thing like a background color. The best approximation is what I showed you.

**Comment by meghanaviyyapu** [comment]
How to get RGB value from  pix.color_topusage().?

**Reply by JorjMcKie** [reply]
the pixel value content

**Reply by JorjMcKie** [reply]
I now have a decent internet connection again. Second item in the  tuple `(ratio, pixel)` is a `bytes` object of length 3 (4) if the Pixmap is RGB(A). So the items of this object represent the integers red, green, blue (, alpha).

**Reply by meghanaviyyapu** [reply]
Thanks so much for the explanation
